code-challenges.test.js

```diff
@@ -14,38 +16,125 @@
+      2, 0, -1, 0,
+    ]);
+    expect(onlyMods([5, "Hola", 43, -34, "greetings", true])).toEqual([
+      2, 1, -1,
+    ]);
+  });
+});
 
 // b) Create the function that makes the test pass.
 
+// 2b) Pseudo Code:
+// Create the function and give it a parameter (array)
+// Use either an HOF (probably .filter) or a for loop to iterate through the input array
+// Iteration logic
+//    Use a method to filter through the datatypes to only return a number
+//    Have the return result be the modulus 3 of the current value
+// Run the test to ensure it passes
+
+const onlyMods = (array) => {
+  let numbersArray = array.filter((value) => {
+    return typeof value === "number";
+  });
+  return numbersArray.map((value) => {
+    return value % 3;
+  });
+};
+
+// Notes: I ran the initial test and got the error. While writing the code, I realized how .filter actually works and that I can't really use mod 3 on the output. I had to create another HOF to take care of that. But overall, not too difficult. I'd say if I could combine these two HOFs into one that would be ideal. Maybe there is a method which could do this? Will check if time permits.
 
 // --------------------3) Create a function that takes in an array of numbers and returns the sum of all the numbers cubed.
 
 // a) Create a test with an expect statement using the variables provided.
 
-const cubeAndSum1 = [2, 3, 4]
-// Expected output: 99
-const cubeAndSum2 = [0, 5, 10]
-// Expected output: 1125
+// 3a) Pseudo Code:
+// Copy and paste test code template into assignment (make sure nothing is inside initial template)
+// Read problem statement and understand inputs/outputs (inputs need to be directly inserted into test code)
+// Create relevant info for describe, it, expect, and toEqual methods
+// Test code output to ensure it is failing red (should get a reference error saying function is not defined)
 
+describe("cubit", () => {
+  it("a function which takes in an array and returns the sum of the cubes of the elements", () => {
+    expect(cubit([2, 3, 4])).toEqual(99);
+    expect(cubit([0, 5, 10])).toEqual(1125);
+  });
+});
 
 // b) Create the function that makes the test pass.
+
+// 3b) Pseudo code:
+// Create the function and give it a parameter (array)
+// Create a variable to store the sum value (initialize to 0)
+// Use iteration to go through each element, cube that value, and sum it to the sum variable
+// Return the sum variable
+// Run the test to ensure it passes
+
+const cubit = (array) => {
+  let sum = 0;
+  array.forEach((value) => {
+    return (sum += value ** 3);
+  });
+  return sum;
+};
+
+// Notes: I ran the intial test and got the right error. This code was really easy to implement. I'd like to refactor it make this a one-liner. It definitely seems possible. However I don't see how I can get away from declaring a variable at the start. It's a pretty concise code as is I guess. But I will explore some form of refactoring if time permits.
```

Comment:
This looks great. You definitely need some kind of anchor outside the iteration. If your goal is just to reduce the number of lines you can one-line the HOF `array.forEach((value) => sum += value ** 3)`